runtime.go

```diff
@@ -2268,29 +2477,34 @@ func (r *Runtime) getIterator(obj Value, method func(FunctionCall) Value) *Objec
 	}
 
 	return r.toObject(method(FunctionCall{
+		ctx:  r.vm.ctx,
 		This: obj,
 	}))
 }
 
 func returnIter(iter *Object) {
 	retMethod := toMethod(iter.self.getStr("return", nil))
 	if retMethod != nil {
-		iter.runtime.toObject(retMethod(FunctionCall{This: iter}))
+		iter.runtime.toObject(retMethod(FunctionCall{ctx: iter.runtime.ctx, This: iter}))
+		// TODO: this was breaking the TestIteratorReturnErrorNested test
```

Comment:
Calling this out as it was quite interesting that this change was causing a very odd error in one of the tests, might require some further investigation unless we're ok with the change.

Reply:
what was the odd error you were seeing?

Reply:
Failing this test `TestIteratorReturnErrorNested` with:

```
panic: runtime error: index out of range [-1] [recovered]
        panic: runtime error: index out of range [-1]
```

This is the one I mentioned on my previous PR and I was finally able to isolate it to this specific line.

Reply:
by adding the ctx, this line fails? or - this line failed because of some changes from upstream's master?

Reply:
Sorry I should have commented a few lines lower:

```go
_ = iter.runtime.tryFunc(func() {
	retMethod(FunctionCall{ctx: iter.runtime.ctx, This: iter})
})
```

instead of 
```go
iter.runtime.toObject(retMethod(FunctionCall{ctx: iter.runtime.ctx, This: iter}))
```

caused made that test fail.